src/libcore/ptr.rs

```diff
@@ -2685,7 +2685,7 @@ impl<T: Sized> NonNull<T> {
     /// This is useful for initializing types which lazily allocate, like
     /// `Vec::new` does.
     #[stable(feature = "nonnull", since = "1.25.0")]
-    pub fn dangling() -> Self {
+    pub const fn dangling() -> Self {
```

Comment:
Please make the constness unstable by adding the `#[rustc_const_unstable]` attribute